integration-provisioning/src/main/java/org/symphonyoss/integration/provisioning/service/ApplicationService.java

```diff
@@ -149,15 +176,32 @@ public boolean updateAppSettings(Application application) {
     String sessionToken = authenticationProxy.getSessionToken(DEFAULT_USER_ID);
 
     try {
-      Map<String, String> app = client.getAppByAppGroupId(appType, DEFAULT_USER_ID);
+      Map<String, Object> app = client.getAppByAppGroupId(appType, DEFAULT_USER_ID);
 
       if (app != null) {
         AppEntitlement appEntitlement = new AppEntitlement();
         appEntitlement.setAppId(appType);
         appEntitlement.setAppName(application.getName());
-        appEntitlement.setEnable(application.isEnabled());
-        appEntitlement.setListed(application.isVisible());
-        appEntitlement.setInstall(application.isAutoInstall());
+
+        if (app.containsKey(APP_SETTINGS)) {
```

Comment:
This code block already exists (lines 135 to 148). Can this be a separated function?

Reply:
Done.